src/Model/Resource.php

```diff
@@ -180,6 +180,33 @@ public static function get($id, $collectionUrl = null, ClientInterface $client)
         }
     }
 
+    /**
+     * Post request for returned data.
+     *
+     * @param string          $body          array of the data to send in the post.
+     * @param string          $collectionUrl The URL of the collection.
+     * @param ClientInterface $client        A suitably configured Guzzle
+     *                                       client.
+     *
+     * @return array The resource object, or false if the resource is
+     *                      not found.
+     */
+    public static function post(array $body, $collectionUrl = null, ClientInterface $client)
```

Comment:
this is mimicking `get()` but I think you want to mimic `getCollection()`, as it's a list of results you're after

you could add a parameter to getCollection to override the request method, `getCollection( ... , $method = 'get')`

Reply:
Would `getCollection` work? Given that is also processes the returned data with `wrapCollection`. 

I thought we might need to implement more `post`s in future if we use the APIs more so it seems reasonable to create a new post method. If you don't think that's the case we can alter an existing method.

Reply:
It takes a $collectionUrl so it seems like you're trying to fetch a collection; list of things. But it's ignoring 404s, which isn't appropriate for a POST nor for a collection.